autoload/codi.vim

```diff
@@ -186,6 +186,34 @@ function! s:user_au(au)
   endif
 endfunction
 
+" If percent / float width, calculate based on buf width
+" If result leaves less than |minwidth| / 20 columns of working space,
+" return a width that leaves |minwidth| / 20 columns to write code in
+" Else, return absolute width as given
+function! s:pane_width()
+  let width = s:get_opt('width')
+  let width = type(width) == type('') ? width : string(width)
+
+
+  if match(width, '[.%]') > -1
+    let raw          = str2float(substitute(width, '%', '', 'g'))
+    let clamped      = raw > 100.0 ? 100.0 : (raw < 0.0 ? 0.0 : raw)
```

Comment:
That said, we can do `let clamped = max(0.0, width)`.

Reply:
`let clamped = max(0.0, width)` will not work. e.g:

![image](https://user-images.githubusercontent.com/3225058/44516420-5636cc00-a6c5-11e8-8f78-8b131fd1ac88.png)

and second version using a list (`let clamped = max([0.0, width])`):

![image](https://user-images.githubusercontent.com/3225058/44516442-664eab80-a6c5-11e8-9acd-f08f142eabc1.png)

Hence the ugly ternary. It looks a lot simpler when only clamping one side:

```viml
let clamped = raw < 0.0 ? 0.0 : raw
```

As a matter of fact, I decided to inline it since it is now quite a bit shorter.